### ISSUE TITLE
Confusing UX when the specified box doesn't support the available providers

### ISSUE THREAD

**Issue by edmorley** [mitchellh/vagrant#6873]
A contributor to our open source project reported that he was unable to `vagrant up` our project, getting (with vagrant v1.7.2 on Fedora):

``` bash
$ vagrant up
Bringing machine 'default' up with 'libvirt' provider...
==> default: Box 'ubuntu/trusty32' could not be found. Attempting to find and install...
    default: Box Provider: libvirt
    default: Box Version: >= 0
==> default: Loading metadata for box 'ubuntu/trusty32'
    default: URL: https://atlas.hashicorp.com/ubuntu/trusty32
The box you're attempting to add doesn't support the provider
you requested. Please find an alternate box or use an alternate
provider. Double-check your requested provider to verify you didn't
simply misspell it.

If you're adding a box from HashiCorp's Atlas, make sure the box is
released.

Name: ubuntu/trusty32
Address: https://atlas.hashicorp.com/ubuntu/trusty32
Requested provider: [:libvirt
```

He'd followed our docs saying to install virtualbox, and our Vagrantfile ([link](https://github.com/mozilla/treeherder/blob/1d1311f3cd7480f4108a40e0d4aa12f1f03a5da6/Vagrantfile)) only specifies options for virtualbox:

``` ruby
  config.vm.provider "virtualbox" do |vb|
    vb.customize ["modifyvm", :id, "--memory", "2048"]
  end
```

The error message above is suboptimal for a few reasons:
1. It says "doesn't support the provider you requested", when the user didn't actually request a provider, Vagrant picked one (presumably since it was the only available provider).
2. It doesn't say which providers the box **does** support.
3. It doesn't say whether those providers were just not installed, or whether they were actually broken/not usable. (In this contributor's case it turned out to be the latter)
4. It doesn't tell the user how to select another provider (ie `--provider`).
5. Given that the box didn't support 'libvirt', it would make more sense to try the providers it does support, and output any errors to the console from them.

Many thanks :-)

**Comment by edmorley** [comment]
When the user re-ran with `--provider=virtualbox`, he got:
```bash
$ vagrant up --provider=virtualbox
The provider 'virtualbox' that was requested to back the machine
'default' is reporting that it isn't usable on this system. The
reason is shown below:
 
VirtualBox is complaining that the kernel module is not loaded. Please
run `VBoxManage --version` or open the VirtualBox GUI to see the error
message which should contain instructions on how to fix this error.
```
Ideally this error message would have been what was output originally.

(Not that it's relevant so much for this ticket, but him then running `VBoxManage --version` gave this...)
```bash
$ VBoxManage --version
WARNING: The vboxdrv kernel module is not loaded. Either there is no module
         available for the current kernel (4.2.8-300.fc23.x86_64) or it failed to
         load. Please recompile the kernel module and install it by
 
           sudo /sbin/rcvboxdrv setup
 
         You will not be able to start VMs until this problem is fixed.
5.0.12_RPMFusionr104815
```

**Comment by chrisroberts** [comment]
Hi there,

Thanks for reporting this bug. I'm sure this was a real issue when originally reported (our fault for not looking sooner!) but there have been multiple Vagrant releases since the original report. I'm going to close this issue now and request you reopen the issue if you're still experiencing this problem. I'm sorry this wasn't looked at earlier :frowning_face: 

Cheers!

**Comment by edmorley** [comment]
The error message in question comes from:
https://github.com/mitchellh/vagrant/blob/3ec6290ad06110b149d87a8403e0f9cff89b0b03/templates/locales/en.yml#L377

Which is used by the `BoxAddNoMatchingProvider` error, which is raised here:
https://github.com/mitchellh/vagrant/blob/9c299a2a357fcf87f356bb9d56e18a037a53d138/lib/vagrant/action/builtin/box_add.rb#L208

The logic in that area hasn't changed since the issue was opened (both from looking at the git log, and by inspecting the current conditionals), so I believe the issue here still exists.

Please could you reopen? :-)

**Comment by vhosakot** [comment]
I saw the same error with vagrant 1.8.7 on CentOS 7.3 with kernel version 3.10.0-514.10.2.el7.x86_64.

```
The vboxdrv kernel module is not loaded. Either there is no module
         available for the current kernel (3.10.0-514.10.2.el7.x86_64)
```

The following steps resolved the error for me:
1. `yum -y install kernel-headers kernel-devel`
2. Reboot.
3. `sudo /sbin/rcvboxdrv setup`
4. `VBoxManage --version`
